nats.go

```diff
@@ -2728,13 +2727,20 @@ func (nc *Conn) isConnected() bool {
 
 // Stats will return a race safe copy of the Statistics section for the connection.
 func (nc *Conn) Stats() Statistics {
-	return Statistics{
-		InMsgs:     atomic.LoadUint64(&nc.InMsgs),
-		InBytes:    atomic.LoadUint64(&nc.InBytes),
-		OutMsgs:    atomic.LoadUint64(&nc.OutMsgs),
-		OutBytes:   atomic.LoadUint64(&nc.OutBytes),
-		Reconnects: atomic.LoadUint64(&nc.Reconnects),
+	// Stats are updated either under connection's mu or subsMu mutexes.
+	// Lock both to safely get them.
+	nc.mu.Lock()
+	nc.subsMu.RLock()
```

Comment:
I created `subsMu` as a `sync.RWMutex` (I could change it to regular `sync.Mutex`). In this case, we are "reading" the stats, not making any modification to either stats or subs map, so read-lock is enough.